mantidproject/mantid
# Bring future.pyplot into the standard MantidPlot namespace

This is for trac ticket [#11542](http://trac.mantidproject.org/mantid/ticket/11542). 

I already updated the course materials: http://www.mantidproject.org/Matrix_and_Plot_Control#Matplotlib-like_plotting_interface, where apparently the qtiplot plot() function is never used. We seem to be lucky enough that it is not used in custom interfaces either (as they seem to use plotSpectrum etc. directly).

**To test:**
- Tests should pass (this includes MantidPlotPyplotGeneralTest which has been modified and renamed).
- Check that the documentation is updated (besides the namespace changes there are a few additions as explained in the ticket).
- Check if you like the namespaces used. I've left both pyplot and qtiplot inside pymantidplot, and I think that it is tidy like that. But there could be other options.
- I tested this with a clean build (as some directories and files are taken out and cmake will not bother to remove them). I think there are no issues with this, but if you're picky and patient you might want to do a clean build.
- you could test a few commands manually just to double-check. You can
  pick examples from the module documentation, like:

``` python
# if this works well, then there's no conflict with qtiplot.plot
plot([0.1, 0.2, 0.3, 0.4], [1.2, 1.3, 0.2, 0.8])
ws = Load("/path/to/MAR11060.raw", OutputWorkspace="foo")
plot(ws, 100)
title('mar spectrum plot)
# should do exactly the same:
plot(ws, 100, tool='sp')

# but qtiplot.plot is still there
pymantidplot.qtiplot.plot('MAR11060', [800, 801])
```
